Common/Securities/CurrencyConversion/ICurrencyPathProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Securities.CurrencyConversion
+{
+    public interface ICurrencyPathProvider
+    {
+        CurrencyEdge AddEdge(string currencyPair, SecurityType type);
+
+        CurrencyEdge AddEdge(string baseCode, string quoteCode, SecurityType type);
```

Comment:
I think we can remove both of these `AddEdge` methods from this interface and instead have the instance properly initialized. IOW, the interface shouldn't concern it'self with defining the edges, since eventually it should support a brokerage getting real time symbol lists or something similar.

I don't think we need all of these 'graph-centric' types, such as edges/vertices, ect...  They're all graph implementation specific and we're not doing a graph solution. Taking a step back, we're trying to solve a currency conversion problem and we want to support the potential for swapping out the implementation later. In order to achieve that goal, we want any abstractions/types to be conversion-centric.

The following is a rough cut at what I think the major objects at work in solving currency conversions. First, we have a definition of a currency conversion.
``` csharp
public interface ICurrencyConversion
{
    string SourceCurrency { get; }
    string DestinationCurrency { get; }
    void Update(); // pokes the component so it can update it's state
    decimal GetConversion(); // gets the current conversion rate
}

// implementation of ICurrencyConversion using Security objects as a data source
public class SecurityCurrencyConversion
{
    private decimal _conversionRate = 1m;
    private readonly List<Security> _securities; //ctor init
    public string SourceCurrency { get; } //ctor init
    public string DestinationCurrency { get; } //ctor init
    public decimal GetConversion() { return _conversionRate; }
    public void Update() { // update _conversionRate via the _securities }

    public static SecurityCurrencyConversion LinearSearch( ...args... ) { // linear search for path }
}
```
Since we only have a single implementation, we don't need to worry about a 'security currency conversion provider' -- we can just call a static factory method on `SecurityCurrencyConversion`, such as `LinearSearch` in the example to keep things as simple as possible for now.

>NOTE: no guarantees the code compiles, github isn't that good... yet!

Reply:
Ok, I am working on this one now!